L1Trigger/TrackerTFP/interface/DataFormats.h

```diff
@@ -0,0 +1,688 @@
+    // stub eta sector
+    int sectorEta() const { return sectorEta_; }
+    // stub radius wrt chosenRofPhi
+    double r() const { return std::get<0>(data_); }
+    // stub phi wrt phi sector centre
+    double phi() const { return std::get<1>(data_); }
+    // stub z residual wrt eta sector
+    double z() const { return std::get<2>(data_); }
+    // reduced layer id
+    int layer() const { return std::get<3>(data_); }
+    // first inv2R bin this stub belongs to
+    int inv2RMin() const { return std::get<4>(data_); }
+    // last inv2R bin this stub belongs to
+    int inv2RMax() const { return std::get<5>(data_); }
+  private:
+    // inv2R bins this stub belongs to
+    TTBV inv2RBins_;
+    // stub phi sector
+    int sectorPhi_;
+    // stub eta sector
+    int sectorEta_;
+  };
+
+  // class to represent stubs generated by process hough transform
+  class StubHT : public Stub<double, double, double, int, int, int, int> {
+  public:
+    // construct StubHT from Frame
+    StubHT(const TTDTC::Frame& frame, const DataFormats* dataFormats, int inv2R);
+    // construct StubHT from StubGP
+    StubHT(const StubGP& stub, int inv2R, int phiT);
+    ~StubHT(){}
+    // stub qOver pt
+    int inv2R() const { return inv2R_; }
+    // stub radius wrt chosenRofPhi
+    double r() const { return std::get<0>(data_); };
+    // stub phi residual wrt track parameter
+    double phi() const { return std::get<1>(data_); };
+    // stub z residual wrt eta sector
+    double z() const { return std::get<2>(data_); };
+    // reduced layer id
+    int layer() const { return std::get<3>(data_); };
+    // phi sector
+    int sectorPhi() const { return std::get<4>(data_); };
+    // eta sector
+    int sectorEta() const { return std::get<5>(data_); };
+    // stub phi at radius chosenRofPhi wrt phi sector centre
+    int phiT() const { return std::get<6>(data_); };
+  private:
+    // fills track id
+    void fillTrackId();
+    // stub qOver pt
+    int inv2R_;
+  };
+
+  // class to represent stubs generated by process mini hough transform
+  class StubMHT : public Stub<double, double, double, int, int, int, int, int> {
+  public:
+    // construct StubMHT from Frame
+    StubMHT(const TTDTC::Frame& frame, const DataFormats* dataFormats);
+    // construct StubMHT from StubHT
+    StubMHT(const StubHT& stub, int phiT, int inv2R);
+    ~StubMHT(){}
+    // stub radius wrt choenRofPhi
+    double r() const { return std::get<0>(data_); }
+    // stub phi residual wrt finer track parameter
+    double phi() const { return std::get<1>(data_); }
+    // stub z rsidual wrt eta sector
+    double z() const { return std::get<2>(data_); }
+    // reduced layer id
+    int layer() const { return std::get<3>(data_); }
+    // phi sector
+    int sectorPhi() const { return std::get<4>(data_); }
+    // eta sector
+    int sectorEta() const { return std::get<5>(data_); }
+    // stub phi at radius chosenRofPhi wrt phi sector centre
```

Comment:
Add comment above pair of lines defining phiT() and inv2R() saying "MHT cell assignment"